docs/subnets/create-a-vm-blobvm.md

```diff
@@ -0,0 +1,470 @@
+# Create a Blob Virtual Machine (VM)
+
+## Introduction
+
+Virtual Machine is a blueprint for a blockchain and a blockchain instantiates from VM, very much similar to a class-object relationship. Issuance of transactions, transaction types, block structure, block building algorithm, keeping pending transactions in the mempool, gossiping transactions to the connected nodes, etc. are defined in a virtual machine.
+
+Virtual Machines can be broken down into components like **Transaction**, **Mempool**, **Network**, **Block**, etc. Each component works together to:
+
+- Define the representation of the blockchain's state
+- Represent the operations in that state
+- Apply the operations in that state
+
+## BlobVM
+
+In this tutorial, we will learn all about virtual machines and how we can build them by taking references from [BlobVM](https://github.com/ava-labs/blobvm). It's a virtual machine that can be used to instantiate key-value blockchains for storing files like images, videos, etc. in an efficient way.
+
+Blobs are small chunks of data. In BlobVM, we divide a file into small blobs and store them as a key-value pair. The key of these small chunks of a file is linked together as the children of the root. This tree is also stored as JSON data against the root's key.
+
+## How to Use a VM
+
+VM modules are served as a binary plugin to the `avalanchego` software. The nodes that want to power their chains using a particular VM, must have the built binary in their `avalanchego`'s `build/plugins` directory (see [here](../nodes/maintain/avalanchego-config-flags.md#--build-dir-string) for more details). There could be multiple VM plugins in this directory.
+
+## High-Level Overview
+
+Virtual Machines are created as a module, whose binary is registered (loaded) by a node running `avalanchego`, against a user-defined **vmID** (binary file name must be vmID). We can build multiple chains with the registered VM. A VM is registered, only when its built binary is present in the `build/plugins/` directory.
+
+The VM exposes 3 things for interaction:
+
+- **Static and Non-Static Handlers (API)** for making requests to VM methods
+- **Service** for implementing handlers or APIs
+- **Client** for making calls to the service's APIs
```

Comment:
3 things for interaction is strange wording to me, it sounds like this is the only purpose of the VM, but it doesn't include the actual logic for the consensus engine.

Can you re-phrase this to make it specific to the APIs exposed by the VM?

Also, the Client is not actually part of the VM interface, we have just implemented a client in the VM packages for calling the same APIs. I think it makes sense to include it in this tutorial, but we should clarify that it's not part of the VM itself.

Reply:
I think this piece could be under a section such as `Implementing APIs for BlobVM`